app/Post_Types/Archive_Item/AdminModifications.php

```diff
@@ -13,6 +17,31 @@ public function hooks() {
 		add_filter( 'admin_body_class', array( &$this,'admin_body_class') );
 		add_filter( 'manage_edit-diviner_archive_item_columns', [ $this, 'archival_item_columns' ] );
 		add_action( 'manage_diviner_archive_item_posts_custom_column', [ $this, 'manage_diviner_archive_item_posts_custom_column' ], 10, 2 );
+		add_action( 'carbon_fields_register_fields', [ $this, 'active_field_setup' ], 3, 0 );
+	}
+
+	function active_field_setup(  ) {
+		// get active fields
+		$field_query = new \WP_Query( array(
+			'post_type' => Diviner_Field::NAME,
+			'meta_query'=> array(
+				array(
+					'key'     => Helper::get_real_field_name(FieldPostMeta::FIELD_ACTIVE ),
+					'value'   => FieldPostMeta::FIELD_CHECKBOX_VALUE
+				),
+			),
+		) );
+		// call setup on active fields
+		while( $field_query->have_posts() ) : $field_query->the_post();
+			$field_type = carbon_get_the_post_meta( FieldPostMeta::FIELD_TYPE, 'carbon_fields_container_field_variables' );
+			$field = Diviner_Field::get_class( $field_type );
+			$static_call_name = sprintf(
+				'%s::setup',
+				$field
+			);
+			call_user_func($static_call_name);
```

Comment:
This code will not be picked up by IDEs as a use; modify it to:

```php
call_user_func(array($field, 'setup'));
```

Reply:
good point